core/src/test/java/org/elasticsearch/common/xcontent/XContentParserUtilsTests.java

```diff
@@ -49,6 +54,76 @@ public void testEnsureExpectedToken() throws IOException {
         }
     }
 
+    public void testParseStoredFieldsValueString() throws IOException {
+        final String value = randomAlphaOfLengthBetween(0, 10);
+        assertParseStoredFieldsValue(value, (xcontentType, result) -> assertEquals(value, result));
+    }
+
+    public void testParseStoredFieldsValueInt() throws IOException {
+        final Integer value = randomInt(1_000);
```

Comment:
nit: can we test the whole positive/negative int range here? I think it shouldn't complicate things a lot and since its testing the parser it would provide better coverage.